dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityModeHandler.java

```diff
@@ -0,0 +1,56 @@
+package com.dotcms.rendering.velocity.servlet;
+
+import com.dotcms.visitor.business.VisitorAPI;
+
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.web.HostWebAPI;
+import com.dotmarketing.business.web.WebAPILocator;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.PageMode;
+
+import java.io.StringWriter;
+import java.io.Writer;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public abstract class VelocityModeHandler {
+    
+    protected String CHARSET = Config.getStringProperty("CHARSET", "UTF-8");
+    protected static HostWebAPI hostWebAPI = WebAPILocator.getHostWebAPI();
+    protected static VisitorAPI visitorAPI = APILocator.getVisitorAPI();
+    
+    
+    
+    abstract void serve() throws Exception;
+
+    abstract void serve(Writer out) throws Exception;
+
+
+    public final String eval() {
+        StringWriter out = new StringWriter(4096);
+        try {
+            serve(out);
+        } catch (Exception e) {
+            throw new DotRuntimeException(e);
+        }
+        return out.toString();
+    }
+
+    public final static VelocityModeHandler modeHandler(PageMode mode, HttpServletRequest request, HttpServletResponse response) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reorder the modifiers to comply with the Java Language Specification. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AModifiersOrderCheck)